pkg/controller/operators/catalog/not_found.go

```diff
@@ -0,0 +1,39 @@
+package catalog
+
+import (
+	"fmt"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+)
+
+// notFound queries API Discovery for GroupVersions that have been notFound and are now removed in kube 1.22+.
+// OLM will return an install plan error if a non-existent APIVersion is found in an installplan step before attempting to create it on-cluster.
+// See https://kubernetes.io/docs/reference/using-api/deprecation-guide/ for notes on notFound resources.
+
+// notFoundError is returned from installplan execution when a step contains a resource in a GroupVersion that is not found on the cluster.
+type notFoundError struct {
+	gv   schema.GroupVersion
+	k    string
+	name string
+}
+
+func (d notFoundError) Error() string {
+	return fmt.Sprintf("server resource not found: %s of Kind %s in GroupVersion %s is not found on this cluster. %s", d.name, d.k, d.gv,
```

Comment:
The error message needs to clarify whether it's the specific resource doesn't or the entire API (GVK).